scripts/protocversioncheck.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+PROTOC_CMD=${PROTOC_CMD:-protoc}
+PROTOC_VERSION_MIN="$1"
+echo "==> Checking that protoc is at version $1..."
+
+PROTOC_VERSION=$($PROTOC_CMD --version | grep -o '[0-9]\+\.[0-9]\+\(\.[0-9]\+\)\?')
+
+IFS="." read -r -a PROTOC_VERSION_ARR <<< "$PROTOC_VERSION"
+IFS="." read -r -a PROTOC_VERSION_REQ <<< "$PROTOC_VERSION_MIN"
+
+if [[ ${PROTOC_VERSION_ARR[0]} -lt ${PROTOC_VERSION_REQ[0]} ||
```

Comment:
Should we just make sure the user has the same protoc version as the minimum? Cause as noted below, if they differ, it would cause the same issue as we are facing now, no?

Reply:
Yes, I want it to match exactly, for the reasons you say. I'll double check this.

Reply:
Fixed in https://github.com/hashicorp/vault/pull/17122/commits/c9996bc54ed22e569a8e71273c9fe91bcb585737